Add Tools directory and files to Feat/dwin nextgen support

Thinkersbluff opened the pull request:
I forked this repository, to help me learn how to work with these tools on a local copy of my own, and as requested in the extui ReadMe.

UPDATE (27 April 2022):
I started this PR started as a "practice-run", by proposing that the DGUS Tools and How-To info be updated on the extui branch.  
It then continued to evolve, by accident  (oops!), when I continued to push my other changes to the same branch on my repository, while I finished updating the work that InsanityAutomation had started. 
At this point, the changes bundled under this PR are ready to consider for a squash-merge back into the main extui branch, with the possible exception that the wonderfully detailed "How To DGUS" knowledge has not been retained and should not be lost.

I recommend that we capture and move that information into the WIKI as its own dedicated section, and that we continue to work on updating and improving it, to lower the barrier for future developers and maintainers of this UI firmware.

Sebazzz committed:
Support for newer DWIN hardware by @InsanityAutomation

Thinkersbluff committed:
Create Tools directory

The link to the DGUS tools used to build versions 5, 6 and 6.1 of this firmware are no longer available at the link provided in the ReadMe file.
Rather than risk losing the ability to repair existing screens, and potentially losing the tools to be used to build fuure releases, this directory shall hold an archived copy of each toolset used to build and maintain all display firmware on this project.

Thinkersbluff committed:
Adds archived (7z) files containing the DGUS tools

DGUS Toolv8.0 was used to build and maintain versions 5 and 6 of the Community display firmware.

DGUS Tool v8.2 will be used to build and maintain the Community display firmware after version 6.1-Final.  It is too large to store as a single file in GitHub, so it has been split into two files.  Download and recombine those two files with 7zip.

Thinkersbluff committed:
Thinkersbluff Update - 20 Apr 22

These files flash correctly to my CR6-SE and run on DGUS2 v4.0 Fastboot.  (See Known Issues text file for an explanation of the two issues that remain in this DRAFT.)

Thinkersbluff committed:
Thinkersbluff Update - 20 Apr 22

These files flash correctly to my CR6-SE and run on DGUS2 v4.0 Fastboot.  (See Known Issues text file for an explanation of the two issues that remain in this DRAFT.)

Thinkersbluff committed:
Delete chinese versions of 13.bin and 14.bin

Otherwise, Build flags them as overlapping files.

Thinkersbluff committed:
Do not bundle kernel upgrade in DWIN_SET folder

"Upgrading" v4.5 displays to v3.5 has been known to "brick" the display.  Better to let the user decide which DGUS kernel to flash when they install the UI.

Thinkersbluff committed:
Add a BUILD folder to DOC

GitHub Desktop did not push build as its own folder.  Wanted to find a way to capture and post DRAFT builds, to facilitate collaboration without tainting the upstream repository

Thinkersbluff committed:
Corrected folder name

Have actually edited/replaced some of the 20 april files in this version.

Thinkersbluff committed:
Update README.md

Tailored this file to this particular fork

Thinkersbluff committed:
Update README.md

Thinkersbluff committed:
Rename README.txt to README.md

Thinkersbluff committed:
23 April DRAFT

Known issues include:
11. Upon completion of Filament Load or Unload:
 - the final message confirming that the process has completed IS NOT displayed
2. The two buttons at the top of the Prepare menu (LED and Sound) are meant to have ON/OFF icons that confirm the currently commanded state of those controls.  Neither ICON is appearing.
3. The LED ON/OFF button on the Printing->Tune screen is actually using the Retract Override icons. The button does control the LED, though.

Thinkersbluff committed:
Post 24 April Draft, Update README.MDs

Updates the Build folder with the latest DRAFT as of 24 April.
Still chasing two "mysteries"
1. The ON/OFF icons are not displaying for LED or Sound.  The Tune menu LED icons were 3 & 4 instead of 1 & 2. At least they used to appear.  I corrected the icon #'s and regenerated. Now no icons appear when the button is toggled.
2. The Filament Load/Unload Completed message used to appear on the control screen, in miniature text. Now it does not appear at all.

Thinkersbluff committed:
Copied 24 April files to GitHub

Overwriting src files to ensure the latest files I am working with are also the ones in this development branch of the repository

Thinkersbluff committed:
Delete T5L_UI_DGSU2_V40_Fast boot_20210917.BIN

Removes kernel upgrade file from DWIN_SET folder.
Users must install the kernel upgrade file of their own choosing, after reviewing the README.MD and considering their particular situation vis-a-vis display hardware.

Thinkersbluff committed:
Update 24 April DRAFT with corrected ON/OFF toggles for LED and Sound

Figured out that the ON/OFF "icons" were actually coded as "Toggles".  Pointed those objects to 32_toggles.icl instead of 45_icons.icl & positioned them correctly.
Updated Known Issues.txt accordingly.

This DRAFT has just two known issues remaining:
1. Upon completion of Filament Load or Unload, the message confirming that the process has completed IS NOT displayed. The M117 message space is instead just blank.

2. The ABLprocess no longer uses the contents of 35_Autolevel.icl to animate the collection & display of the mesh data.  To enable old eyes to read the mesh, the normally white background of the initial screen has been changed to black.  As a result, when the process first initializes all of the values to 0.000 in white, those zeroes are now visible to the user. This will not look familiar to past users of 6.1, but should not be a problem for new users.

Thinkersbluff committed:
Add v4.5 and v3.5 DGUS kernel files to upgrades folder

For convenience of users who wish to experiment.

Thinkersbluff committed:
Update README.md

Added bold text to end of Recommendations section.

Thinkersbluff committed:
Update README.md

Thinkersbluff committed:
Update all Project Files to 24 April 2022 DRAFT

Update all Project Files to those from which 24 April 2022 DRAFT build was built

Thinkersbluff committed:
Update README.md

Thinkersbluff committed:
Update README.md

Thinkersbluff committed:
Update README.md

Fix typo

Thinkersbluff committed:
Update README.md

Refined the recommendations and clarified the Example

Thinkersbluff committed:
Update README.md

Refined the Background section.

Thinkersbluff committed:
Update README.md

Thinkersbluff committed:
Update README.md

Thinkersbluff committed:
Update README.md

Corrected a typo

Thinkersbluff committed:
Update README.md

Thinkersbluff committed:
Update README.md

Thinkersbluff committed:
Update README.md

Thinkersbluff committed:
Update README.md

Added a copy of the T5LCFG_Calibrate_Touchscreen.CFG file to the repository and updated these instructions.

Thinkersbluff committed:
Create T5LCFG_Calibrate_Touchscreen.cfg

Uploaded a CFG file to use

Thinkersbluff committed:
Update README.md

Moved and annotated the DWIN display calibration instructions into their own file, to avoid confusing users with them.

Thinkersbluff committed:
Moved display calibration instructions to their own folder

Created a "FOR DEVELOPERS AND TESTERS ONLY" folder, to avoid suggesting to other users that any sort of display calibration is required.  These instructions are purely intended to help developers and testers who would like to verify UI functionality under both v3.5 and v4.5+

Thinkersbluff committed:
Update KNOWN ISSUES.txt

Added an observation that the LED ON/OFF button is slightly out of position on the Printing->Tune screen. That will be fixed in the next DRAFT.

Thinkersbluff committed:
Update note on Screen 78

The note says assumes a hot-end capable of 6mm retraction, but the code actually commands 4mm of retraction.  A future release will enable users to modify this value by editing the value of G26_RETRACT_MULTIPLIER in Configuration.h.
At that point, this note should be reviewed and probably should be revised to whatever value is baked into the released compiled firmware.

Thinkersbluff committed:
Uploading latest nextgen UI firmware

The LED ON/OFF button on the Printing->Tune menu has now been nudged to the correct location.
The M117 Scrolling Text field is not displaying messages on the screen, but I have not yet deduced the why not.
One way to solve the problem may be to ensure that all messages are less than 29 characters in length, which will then be displayed to the M117 Static text field.

Thinkersbluff committed:
Uploaded Release Candidate dated 26 April 2022

Solved the scrolling message problem, after reviewing this YouTube DWIN tutorial: https://www.youtube.com/watch?v=7Zd0995evaU

Thinkersbluff committed:
Uploading source files for 26 April 2022 Release Candidate

Sebazzz commented on README.md:
```diff
@@ -1,154 +1,23 @@
 # CR-6 Touchscreen software
-Attempt to extend the CR-6 touch screen software. This repository belongs to the [forked Marlin firmware repository](https://github.com/CR6Community/Marlin/).
+Attempt to extend the CR-6 touch screen software to operate with newer display hardware being sold with CR6 machines as of approximately Dec 2021.
```
We might want to clarify here that the firmware is also still valid for "old models".

Sebazzz commented on README.md:
```diff
@@ -1,154 +1,23 @@
 # CR-6 Touchscreen software
-Attempt to extend the CR-6 touch screen software. This repository belongs to the [forked Marlin firmware repository](https://github.com/CR6Community/Marlin/).
+Attempt to extend the CR-6 touch screen software to operate with newer display hardware being sold with CR6 machines as of approximately Dec 2021.
+This repository belongs to Thinkersbluff and is forked from [the CR6Community Touchscreen Firmware repository](https://github.com/CR6Community/CR-6-touchscreen).
```
Let's remove this before merging back 🤓

Sebazzz commented on README.md:
```diff
@@ -1,154 +1,23 @@
-Please find official releases in the [Releases section](https://github.com/CR6Community/CR-6-touchscreen/releases). Take the release which belongs to the particular firmware you are going to flash.
-
-Ensure you take the right assets: the `CR-6-touchscreen-[date].zip`. You should not download the `Source code` archive if you are downloading with the purpose of flashing your LCD.
+This project is not yet ready to release a stable version of the updated display firmware.
```
I propose we merge this back to master as this is the only way forward, given the binary nature of the project,.

Thinkersbluff replied:
There is a lot of detailed info in the current extui baseline README about how to contribute changes to this firmware.
I found it very helpful when I was working with it, but I deleted it completely from the README in the feat branch.
If we merge this branch back as-is, will all of that good info disappear?  I was hoping to keep it but update it to point to the new tools & to add any new instructions that may be required, but we could also create a separate document "For Developers Only" and plan to refine that in a follow-on PR.

Sebazzz commented on doc/build/DRAFT_21 Apr 2022/CR-6 kernel upgrade/README.txt:
```diff
@@ -0,0 +1,14 @@
+
+NOTE: 
+Based on CF User feedback, we believe the following to be true:
+ - Legacy display hardware (Dec 2021 or earlier) does not run on v4.5 or v4.7
```
We need to refine this to make sure and also include instructions on how to find the current version of the display firmware (the empty DWIN_SET folder I believe)

Sebazzz replied:
And of course incorporate this back into the main README.

Sebazzz commented:
LGTM so far!

Let's refine the README and flashing instructions a bit, remove the DRAFT files (so the repository size is less inflated - let's squash-commit-merge this so those files won't exist in the end result), and then I think we can offer these files as part of a 6.1b beta release.

Thinkersbluff committed:
Remove DRAFT builds

Not needed in the PR

Thinkersbluff commented:
So this is part of what I mean when I say I am not very GitHub literate.  

I did not realize that continuing to commit changes to the same branch on my fork would result in a continuing evolution of the PR.

I have removed the DRAFT folders.

I am now refining the README.MD file in the RC folder.
Are there specific points you would like me to add/delete/modify?

Thinkersbluff committed:
Split the README into basic instructions + a Troubleshooting File

I think this covers the rest of your request.

Thinkersbluff committed:
Add units to Setup->Temperature->Bed Temperature field

Units character field was blank.  The Units text was instead in the Boot Text field.
Moved 'C to the units character field and changed the Boot Text to 0.

Thinkersbluff committed:
Update 26 April Release Candidate to v2

Added 'C to Bed Temperature field on screen#48

Updated 13.bin, 14.bin, 22.bin in DWIN_SET.

Renamed folder for 26 April 2022 Release Candidate to add " v2"

Thinkersbluff committed:
Update README.md

Elaborated the credits section to emphasize the foundational role of the core CR6 Community FIrmware dev team.

Sebazzz commented:
The most important thing to clarify is the DWIN OS upgrade. 

As for the rest, check my remarks in the "files" tab.

Thinkersbluff committed:
Create RefactoredCF6.1TouchscreenFirmware-RC_26042022_v2.zip

Thinkersbluff committed:
Update README.md

Incorporates review feedback

Thinkersbluff committed:
Update README.md

Thinkersbluff commented:
I believe this conflicting part of the PR has been overtaken by events.
As long as the latest TFT files are kept in the merge, we will be ok ignoring these ones.

Sebazzz commented:
I see what the problem with merging is. This PR wants to merge to the feature branch instead of extui, but we should merge to extui instead.

Sebazzz commented:
Changed to extui.
